### ISSUE TITLE
meshlabserver doesn't load on macOS - dyld: Library not loaded: @rpath/QtOpenGL.framework/Versions/5/QtOpenGL

### ISSUE THREAD

**Issue by mikeroberts3000** [cnr-isti-vclab/meshlab#64]
Hi there,

After upgrading to Meshlab 2016, I get the following error when I try to run meshlabserver from the commandline.

```
DN0a22a835:MacOS mike$ cd /Applications/meshlab.app/Contents/Frameworks
DN0a22a835:Frameworks mike$ pwd
/Applications/meshlab.app/Contents/Frameworks
DN0a22a835:Frameworks mike$ ls
QtCore.framework		QtOpenGL.framework		QtXml.framework
QtDBus.framework		QtPrintSupport.framework	QtXmlPatterns.framework
QtGui.framework			QtScript.framework		libomp.dylib
QtNetwork.framework		QtWidgets.framework
DN0a22a835:Frameworks mike$ /Applications/meshlab.app/Contents/MacOS/meshlabserver
dyld: Library not loaded: @rpath/QtOpenGL.framework/Versions/5/QtOpenGL
  Referenced from: /Applications/meshlab.app/Contents/MacOS/meshlabserver
  Reason: image not found
Abort trap: 6
DN0a22a835:Frameworks mike$ 
```

Note that I am in the Frameworks folder in this example, so I am already applying the workaround suggested in this thread:

https://sourceforge.net/p/meshlab/discussion/499532/thread/0baf5833/

Note also that I can run a previous version of meshlabserver without any issues.

```
DN0a22a835:Frameworks mike$ pwd
/Applications/meshlab.app/Contents/Frameworks
DN0a22a835:Frameworks mike$ /Applications/meshlab_old.app/Contents/MacOS/meshlabserver

Usage:
    meshlabserver arg1 arg2 ...  
```

Note that in this second example, I'm running the meshlabserver executable in the __meshlab_old.app__ folder, without needing to be in any particular folder.

**Comment by johnwquarles** [comment]
I was attempting to see if Meshlab 2016 still has this issue:

https://sourceforge.net/p/meshlab/discussion/499533/thread/e2cd0545/

and ran into the exact same thing.

**Comment by atatb** [comment]
According to `otool -l meshlabserver`, `@rpath` is used but not specified, so as a temporary workaround, you could do like this.

```
/Applications/meshlab.app/Contents/MacOS: $ install_name_tool -add_rpath "@executable_path/../Frameworks" meshlabserver
```

ref:
http://stackoverflow.com/questions/9263256/can-you-please-help-me-understand-how-mach-o-libraries-work-in-mac-os-x

**Comment by jabooth** [comment]
@atatb's fix seems to be the correct solution here - can the build script be updated to run `install_name_tool` so users don't run into this issue?

**Comment by markusmaresch** [comment]
# The following (and equivalent) worked for me (tested in 2017/10)
# OsX: 10.12.6
# meshlab v2016.12)
cd /Applications/meshlab.app/Contents/MacOS; 
export DYLD_FRAMEWORK_PATH=../Frameworks;
./meshlabserver

**Comment by Agona-Asafo** [comment]
Hello... I use a MacOS High Sierra Ver 10.13.6 and have downloaded and installed meshLab 2016.12. My issue is, it will not run, i.e. when I click on the App in my Applications, it will not run.
Please is there anything I have done wrong or I am not doing as I had assumed it was an easy install and run. I am not a nerd and do not understand all the command line info/instructions that I have been looking at on the discussions boards.
Please help...Thank you.. Agona

**Comment by alemuntoni** [comment]
Hi everyone!
This problem should be fixed with commit [e8a4b95](https://github.com/cnr-isti-vclab/meshlab/commit/e8a4b955c1bf1bb98474b99dfebd1eea4ae04863).
Inside of the github actions job of that commit ([github actions](https://github.com/cnr-isti-vclab/meshlab/commit/e8a4b955c1bf1bb98474b99dfebd1eea4ae04863/checks?check_suite_id=406590351)) you can find an artifact containing a .dmg of meshlab 2020.01. You can install it and test if `meshlabserver` is working.
Therefore, the next release (2020.02) will contain a working `meshlabserver`.
